config/config.go

```diff
@@ -46,42 +46,55 @@ const Mainnet protocol.NetworkID = "mainnet"
 const GenesisJSONFile = "genesis.json"
 
 // Local holds the per-node-instance configuration settings for the protocol.
+// !!! WARNING !!!
+//
+// These versioned struct tags need to be maintained CAREFULLY and treated
+// like UNIVERSAL CONSTANTS - they should not be modified once committed.
+//
+// New fields may be added to the Local struct, along with a version tag
+// denoting a new version. When doing so, also update the
+// installer/config.json.example and test/testdata/configs/config-v{n}.json
+//
+// !!! WARNING !!!
 type Local struct {
 	// Version tracks the current version of the defaults so we can migrate old -> new
 	// This is specifically important whenever we decide to change the default value
 	// for an existing parameter.
-	Version uint32
+	Version uint32 `version[0]:"0" version[1]:"1" version[2]:"2" version[3]:"3" version[4]:"4" version[5]:"5" version[6]:"6" version[7]:"7"`
```

Comment:
Is this strictly necessary, or would it be possible to have use `getLatestConfigVersion` to give the latest version number and fill in the missing versions on initialization?

Reply:
What was the rationale behind putting defaults in tags and not in a map which is used to fill out the configs at initialization? One unfortunate fact about this design is that it's harder to see what fields changed in a particular version. Was the intent here to make it easier to group up changes happening to a single field?

Reply:
Yes, as you've figured, I have tried to place it all in "one" place.